Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterApp.kt

```diff
@@ -0,0 +1,231 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetcaster.tv.ui
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Home
+import androidx.compose.material.icons.filled.Person
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material.icons.filled.Settings
+import androidx.compose.material.icons.filled.VideoLibrary
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.tv.material3.ExperimentalTvMaterial3Api
+import androidx.tv.material3.Icon
+import androidx.tv.material3.MaterialTheme
+import androidx.tv.material3.NavigationDrawer
+import androidx.tv.material3.NavigationDrawerItem
+import androidx.tv.material3.Text
+import com.example.jetcaster.tv.ui.discover.DiscoverScreen
+import com.example.jetcaster.tv.ui.library.LibraryScreen
+import com.example.jetcaster.tv.ui.podcast.PodcastScreen
+import com.example.jetcaster.tv.ui.podcast.PodcastScreenViewModel
+import com.example.jetcaster.tv.ui.profile.ProfileScreen
+import com.example.jetcaster.tv.ui.search.SearchScreen
+import com.example.jetcaster.tv.ui.settings.SettingsScreen
+
+@OptIn(ExperimentalTvMaterial3Api::class, ExperimentalComposeUiApi::class)
+@Composable
+fun JetcasterApp(jetcasterAppState: JetcasterAppState = rememberJetcasterAppState()) {
+    Route(jetcasterAppState = jetcasterAppState)
+}
+
+internal data object JetcasterAppDefaults {
+    val overScanMargin = OverScanMarginSettings()
+    val gapSettings = GapSettings()
+    val cardWidth = CardWidth()
+    val padding = PaddingSettings()
+}
+
+data class OverScanMarginSettings(
+    val default: OverScanMargin = OverScanMargin(),
+    val podcastDetails: OverScanMargin = OverScanMargin(top = 40.dp, bottom = 40.dp),
+    val drawer: OverScanMargin = OverScanMargin(start = 0.dp, end = 0.dp),
+    val catalog: OverScanMargin = OverScanMargin(start = 0.dp, end = 0.dp)
+)
+
+data class OverScanMargin(
```

Comment:
Nit: Consider moving this under a package for themes. Perhaps a new file called `Spacing.kt` under `com.example.jetcaster.tv.ui.theme`?

Reply:
That sounds good. Thanks for the suggestion.